LICENSE

```diff
@@ -403,261 +403,5 @@
-INFRINGE ANY THIRD PARTY RIGHTS.
-
-5. CNRI SHALL NOT BE LIABLE TO LICENSEE OR ANY OTHER USERS OF PYTHON
-1.6.1 FOR ANY INCIDENTAL, SPECIAL, OR CONSEQUENTIAL DAMAGES OR LOSS AS
-A RESULT OF MODIFYING, DISTRIBUTING, OR OTHERWISE USING PYTHON 1.6.1,
-OR ANY DERIVATIVE THEREOF, EVEN IF ADVISED OF THE POSSIBILITY THEREOF.
-
-6. This License Agreement will automatically terminate upon a material
-breach of its terms and conditions.
-
-7. This License Agreement shall be governed by the federal
-intellectual property law of the United States, including without
-limitation the federal copyright law, and, to the extent such
-U.S. federal law does not apply, by the law of the Commonwealth of
-Virginia, excluding Virginia's conflict of law provisions.
-Notwithstanding the foregoing, with regard to derivative works based
-on Python 1.6.1 that incorporate non-separable material that was
-previously distributed under the GNU General Public License (GPL), the
-law of the Commonwealth of Virginia shall govern this License
-Agreement only as to issues arising under or with respect to
-Paragraphs 4, 5, and 7 of this License Agreement.  Nothing in this
-License Agreement shall be deemed to create any relationship of
-agency, partnership, or joint venture between CNRI and Licensee.  This
-License Agreement does not grant permission to use CNRI trademarks or
-trade name in a trademark sense to endorse or promote products or
-services of Licensee, or any third party.
-
-8. By clicking on the "ACCEPT" button where indicated, or by copying,
-installing or otherwise using Python 1.6.1, Licensee agrees to be
-bound by the terms and conditions of this License Agreement.
-
-        ACCEPT
-
-
-CWI LICENSE AGREEMENT FOR PYTHON 0.9.0 THROUGH 1.2
---------------------------------------------------
-
-Copyright (c) 1991 - 1995, Stichting Mathematisch Centrum Amsterdam,
-The Netherlands.  All rights reserved.
-
-Permission to use, copy, modify, and distribute this software and its
-documentation for any purpose and without fee is hereby granted,
-provided that the above copyright notice appear in all copies and that
-both that copyright notice and this permission notice appear in
-supporting documentation, and that the name of Stichting Mathematisch
-Centrum or CWI not be used in advertising or publicity pertaining to
-distribution of the software without specific, written prior
-permission.
-
-STICHTING MATHEMATISCH CENTRUM DISCLAIMS ALL WARRANTIES WITH REGARD TO
-THIS SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND
-FITNESS, IN NO EVENT SHALL STICHTING MATHEMATISCH CENTRUM BE LIABLE
-FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
-WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
-ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT
-OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+See https://github.com/apache/beam/blob/master/LICENSE.python for additional
```

Comment:
Should this be linked to a specific commit?

Reply:
Good question.

If yes, then if LICENSE.python changes, we need to update it. I'd also need to do an inconvenient bit of merging etc to get the commit address properly. (Or simply update the link in a subsequent PR, linking to the previous squashed commit).

I could "cheat" by simply referring to the adjacent LICENSE.python file, if present. That handles images which don't have it included (since they don't include the python interpreter). But I don't know if that's sufficient legally. It's about as durable as any of the LICENSE inclusions we do make though. This would be my preference though, avoids additional maintenance, and keeps the files "paired" as needed.

Reply:
Referring to the adjacent files makes sense to me too. Similar to you, I do not know if it is the right way to do things. When in doubt, you can ask at LEGAL jira.

Reply:
I've made the change, and asked at https://issues.apache.org/jira/browse/LEGAL-582

Reply:
Reading https://infra.apache.org/licensing-howto.html more closely, I posit that this move is probably fine (things are bundled and depended on). The only improvement/change I would make is instead appending the contents of LICENSE.python to LICENSE on the BUILD instead of having the side car file. The main thing about the LICENSE seems to be are whether we're bundling/distributing the code (or whether the dependency in question requires a usage notice license).

The current move seems to more in line with the guidance in https://infra.apache.org/licensing-howto.html since we're only including the CPython license when including CPython, rather than over including it.

What do you think? I think we should wait at most a week for LEGAL to reply back before moving forward. Otherwise I can see what I can do about concatenating LICENSE.python to LICENSE for those containers.

Reply:
That plan sounds good to me.

Reply:
Someone from Legal has replied, saying it's fine. I'll wait for tomorrow before merging in case of further discussion.

Reply:
Further discussion was a 2nd person saying it's fine, and then closing the issue. Marking this as resolved and ready to review.